ports/esp32/boards/GENERIC/mpconfigboard.h

```diff
@@ -1,2 +1,6 @@
-#define MICROPY_HW_BOARD_NAME "ESP32 module"
+#ifndef MICROPY_HW_BOARD_NAME
+// Can be set by mpconfigboard.cmake.
+#define MICROPY_HW_BOARD_NAME "Generic ESP32 module"
+#endif
+
 #define MICROPY_HW_MCU_NAME "ESP32"
```

Comment:
Should this be configured per-variant?  It previously did have different values for the variants, but arguably it should be the same for all of them, so probably OK to keep it at just a single value like you've done.

Reply:
Yeah, so the d2wd and unicore boards did have different names before.

I think you could make the case that d2wd and unicore are attributes of the MCU whereas spiram and ota are attributes of the board/configuration. I think we should fix it.

In theory you could have e.g. a spiram+unicore variant, in which case I think setting the MCU to "esp32-unicore" and the board to "Generic ESP32 module with SPIRAM" makes sense.

Updated.